subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/intra_vc_scheduler.go

```diff
@@ -28,22 +28,42 @@ import (
 	"k8s.io/klog"
 )
 
-// intraVCScheduler is an interface for allocating cells inside a VC.
-// It stores two maps of ChainCellList, one for reserved cells, the other
-// for non-reserved ones. It should be able to return a virtual cell
-// on receiving a cell request.
+// intraVCScheduler is an interface for scheduling pods inside a VC.
+// It stores two maps of ChainCellList, one for reserved cells, the other for non-reserved ones.
+// It should be able to return a set of GPU placements in the VC for a scheduling request.
 type intraVCScheduler interface {
 	getNonReservedCellList() map[CellChain]ChainCellList
 	getReservedCellList() map[api.ReservationId]ChainCellList
 
-	// Cell allocation inside a VC. We use buddy alloc for default
-	// (see defaultIntraVCScheduler).
-	allocateCell(CellRequest) *VirtualCell
+	// Scheduling a series of pods inside a VC. We use topologyAwareScheduler by default.
+	schedule(schedulingRequest) map[int32][]CellList
 }
 
 type defaultIntraVCScheduler struct {
 	virtualNonReservedCellList map[CellChain]ChainCellList
 	virtualReservedCellList    map[api.ReservationId]ChainCellList
+	schedulersNonReserved      map[CellChain]*topologyAwareScheduler
+	schedulersReserved         map[api.ReservationId]*topologyAwareScheduler
```

Comment:
What is the benefits to make a dedicated scheduler for each CellChain and ReservationId?
Why not share the same one?

Reply:
The scheduler operates on a cluster view. If we use the same one for all chains, it needs to calculate the cluster view every time the scheduler is invoked

---
In reply to: [332510858](https://github.com/microsoft/pai/pull/3684#discussion_r332510858) [](ancestors = 332510858)

Reply:
A group may need to relax to across multiple chains.
Will consider in future.

---
In reply to: [334218248](https://github.com/microsoft/pai/pull/3684#discussion_r334218248) [](ancestors = 334218248,332510858)